document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreTest.java

```diff
@@ -334,12 +338,209 @@ public void testWithDifferentFieldTypes(String dataStoreName) throws Exception {
     filter = new Filter(Op.GT, "name", "abc1");
     queryStringField.setFilter(filter);
-    results = collection.search(queryBooleanField);
+    results = collection.search(queryStringField);
     documents = new ArrayList<>();
     while (results.hasNext()) {
       documents.add(results.next());
     }
-    Assertions.assertEquals(2, documents.size());
+    Assertions.assertEquals(3, documents.size());
+
+    datastore.deleteCollection(COLLECTION_NAME);
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void testExistsOperatorWithFind(String dataStoreName) throws Exception {
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    datastore.createCollection(COLLECTION_NAME, null);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+
+    // size field with integer value, isCostly boolean field
+    collection.upsert(
+        new SingleValueKey("default", "testKey1"),
+        Utils.createDocument(
+            ImmutablePair.of("id", "testKey1"),
+            ImmutablePair.of("name", "abc1"),
+            ImmutablePair.of("size", -10),
+            ImmutablePair.of("isCostly", false)));
+
+    collection.upsert(
+        new SingleValueKey("default", "testKey2"),
+        Utils.createDocument(
+            ImmutablePair.of("id", "testKey2"),
+            ImmutablePair.of("name", "abc2"),
+            ImmutablePair.of("size", -20),
+            ImmutablePair.of("isCostly", false)));
+
+    collection.upsert(
+        new SingleValueKey("default", "testKey3"),
+        Utils.createDocument(
+            ImmutablePair.of("id", "testKey3"),
+            ImmutablePair.of("name", "abc3"),
+            ImmutablePair.of("size", 5),
+            ImmutablePair.of("isCostly", true)));
+
+    collection.upsert(
+        new SingleValueKey("default", "testKey4"),
+        Utils.createDocument(
+            ImmutablePair.of("id", "testKey4"),
+            ImmutablePair.of("name", "abc4"),
+            ImmutablePair.of("size", 10),
+            ImmutablePair.of("isCostly", true)));
+
+    org.hypertrace.core.documentstore.query.Query.QueryBuilder queryBuilder;
+    // query exists
+    queryBuilder = org.hypertrace.core.documentstore.query.Query.builder();
+    queryBuilder.setFilter(
+        RelationalExpression.of(
+            IdentifierExpression.of("name"),
+            RelationalOperator.EXISTS,
+            ConstantExpression.of("true")));
+    Iterator<Document> results = collection.find(queryBuilder.build());
+    List<Document> documents = new ArrayList<>();
+    while (results.hasNext()) {
+      documents.add(results.next());
+    }
+    Assertions.assertEquals(4, documents.size());
+
+    queryBuilder = org.hypertrace.core.documentstore.query.Query.builder();
+    queryBuilder.setFilter(
+        RelationalExpression.of(
+            IdentifierExpression.of("junkKey"),
+            RelationalOperator.EXISTS,
+            ConstantExpression.of("true")));
```

Comment:
Nit:
```suggestion
            ConstantExpression.of(true)));
```

Reply:
There are two different sets, one takes string as input and other boolean expression.